torch/_functorch/aot_autograd.py

```diff
@@ -1114,11 +1118,11 @@ def dispatch_and_compile():
     else:
         compiled_fn = dispatch_and_compile()
 
-    if isinstance(mod, torch._dynamo.utils.GmWrapper):
-        # This function is called by the flatten_graph_inputs wrapper, which boxes
-        # the inputs so that they can be freed before the end of this scope.
-        # For overhead reasons, this is not the default wrapper, see comment:
-        # https://github.com/pytorch/pytorch/pull/122535/files#r1560096481
+    if isinstance(
+        mod, torch._dynamo.utils.GmWrapper
+    ) or torch._dynamo.utils.is_compiled_autograd_gm(mod):
```

Comment:
what's the new invariant now? Are these two conditions always guaranteed to be both true / both false?

Reply:
compiled autograd is always going to fail the GmWrapper check

non-compiled autograd graphs with bumpy inputs will pass the GmWrapper check and always fail the second is_compiled_autograd_gm check